### PR TITLE
Add custom user model and refactoring

### PR DESCRIPTION
**Pull request by nikriek**

### CODE REVIEW

File: users/tests/test_serializers.py [hunk 1]
```diff
@@ -25,9 +25,11 @@ def test_create(self):
         """
         Tests that a user gets created when the serializer is initialized with valid data
         """
-        data = {'username': 'ausername2348353', 'password': 'fsfsanbshfbsjkf'}
+        data = {'username': 'ausername2348353', 'password': 'fsfsanbshfbsjkf', 'color': '#ffffff',
+                'full_name': 'a_name'}
         serializer = UserSerializer(data=data)
-
+        serializer.is_valid()
+        print(serializer.errors)
```
Comment (nikriek) [inline users/tests/test_serializers.py]: Remove

File: users/migrations/0001_initial.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                ('last_login', models.DateTimeField(blank=True, null=True, verbose_name='last login')),
+                ('is_superuser', models.BooleanField(default=False, help_text='Designates that this user has all permissions without explicitly assigning them.', verbose_name='superuser status')),
+                ('username', models.CharField(error_messages={'unique': 'A user with that username already exists.'}, help_text='Required. 150 characters or fewer. Letters, digits and @/./+/-/_ only.', max_length=150, unique=True, validators=[django.contrib.auth.validators.UnicodeUsernameValidator()], verbose_name='username')),
+                ('first_name', models.CharField(blank=True, max_length=30, verbose_name='first name')),
+                ('last_name', models.CharField(blank=True, max_length=30, verbose_name='last name')),
+                ('email', models.EmailField(blank=True, max_length=254, verbose_name='email address')),
+                ('is_staff', models.BooleanField(default=False, help_text='Designates whether the user can log into this admin site.', verbose_name='staff status')),
```
Comment (nicoknoll) [inline users/migrations/0001_initial.py:32]: which admin site?

File: users/models.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from django.contrib.auth.models import AbstractUser
+from django.db import models
+from django.core.validators import RegexValidator
+
+
+class User(AbstractUser):
+    full_name = models.CharField(max_length=100)
+    color = models.CharField(max_length=7, validators=[
+        RegexValidator(
+            regex='^#([0-9a-f]{6}|[0-9a-f]{3})$',
```
Comment (nicoknoll) [inline users/models.py]: ^#(?:[0-9a-fA-F]{3}){1,2}$

File: users/models.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+            message='Is not a valid color code',
+        ), # TODO: Test Regex
```
Comment (nicoknoll) [inline users/models.py:12]: Done :)

File: users/migrations/0001_initial.py (hunk 2 above)
Comment (nikriek) [reply users/migrations/0001_initial.py:32]: Django has an admin site included, which we don't use right now. `is_staff` comes from auto migration

File: users/models.py (hunk 4 above)
Comment (nikriek) [reply users/models.py:12]: I mean a unit test for the validation

Comment (nicoknoll) [reply users/models.py:12]: k

File: users/migrations/0001_initial.py (hunk 2 above)
Comment (nicoknoll) [reply users/migrations/0001_initial.py:32]: k

**Review by nicoknoll — APPROVED** [review]

**Pull request merged**